### DISCUSSION TITLE
Half-occupied units not rendered any more?

### DISCUSSION THREAD

**Discussion by stavr666** [netbox-community/netbox#13548] [Q&A]
There was #4170 FR for 2.7.6 version. When this feature was dropped? Or it's just a bug, no one care about?

**Comment by candlerb** [comment]
That feature was merged in commit ce89fa74, and just tweaked the colours.

By "half-occupied" we are talking half-depth devices: mounted front but not rear, or vice versa.  Checking with v3.5.8, it does indeed appear to have broken:

![image](https://github.com/netbox-community/netbox/assets/44789/4ccc6a17-87cf-427c-a3fc-74028dd26fb1)

The rear elevation has `<rect class="slot" ...>` for each unit. If I edit the them manually to `<rect class="slot blocked" ...>` it renders correctly:

![image](https://github.com/netbox-community/netbox/assets/44789/a5b13be1-c8e0-4e11-acd7-a0199c3bf8c0)

(or with class "occupied" it renders as grey rather than pink)

I think you could raise this as a bug report.

EDIT: the last change I see in this area is fc02e15f, and a refactoring in 64080e80, both of which arrived in v3.3.0

**Reply by candlerb** [reply]
By adding some dummy classes, I can confirm that the rectangle with `class="slot"` in the rear elevation comes from `draw_background()` [here](https://github.com/netbox-community/netbox/blob/v3.5.8/netbox/dcim/svg/racks.py#L288).

**Reply by candlerb** [reply]
I remember now: I had raised this before: see #6961.

Half-occupied units were invisible even as far back as v2.11.11, and there was also confusion around what exactly what the CSS classes "blocked" and "occupied" were intended to be used for.

My best guess: "blocked" is when you have a full-depth device (but no rear image), and "occupied" is when you don't have rights to see what is in there.
